ingestion/src/metadata/ingestion/source/dashboard/looker/bulk_parser.py

```diff
@@ -0,0 +1,148 @@
+    metaclass because it is best suited for this purpose.
+    """
+
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        """
+        Possible changes to the value of the `__init__` argument do not affect
+        the returned instance.
+        """
+        if cls not in cls._instances:
+            instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = instance
+        return cls._instances[cls]
+
+
+class BulkLkmlParser(metaclass=SingletonMeta):
+    """
+    Parses and caches the visited files & views.
+
+    Here we'll just parse VIEWs, as we are already getting
+    the explores from the API.
+
+    The goal is to make sure we parse files only once
+    and store the results of the processed views.
+
+    We want to parse the minimum number of files each time
+    until we find the view we are looking for.
+
+    Approach:
+    When we parse, we parse all files *.view.lkml to get all view and cached them. It can speed up the process and avoid
+    infinity loop when parsing includes.
+    """
+
+    def __init__(self, reader: LocalReader):
+        self._views_cache: Dict[ViewName, LookMlView] = {}
+        self._visited_files: Dict[Includes, List[Includes]] = {}
+
+        # To store the raw string of the lkml explores
+        self.parsed_files: Dict[Includes, str] = {}
+        self.parsed_view: Dict[str, List[Includes]] = {}
+
+        self.reader = reader
+        self.__parse_all_views()
+
+    def __parse_all_views(self):
+        file_paths = self.reader.get_local_files(search_key=".view.lkml")
+        for _path in file_paths:
+            file = self._read_file(Includes(_path))
+            lkml_file = LkmlFile.parse_obj(lkml.load(file))
+            self.parsed_files[Includes(_path)] = file
+            for view in lkml_file.views:
+                view.source_file = _path
+                self._views_cache[view.name] = view
+
+    def _read_file(self, path: Includes) -> str:
+        """
+        Read the LookML file
+        """
+        suffixes = Path(path).suffixes
+
+        # Check if any suffix is in our extension list
+        if not set(suffixes).intersection(set(EXTENSIONS)):
+            for suffix in EXTENSIONS:
+                try:
+                    return self.reader.read(path + suffix)
+                except ReadException as err:
+                    logger.debug(f"Error trying to read the file [{path}]: {err}")
+
+        else:
+            return self.reader.read(path)
+
+        raise ReadException(f"Error trying to read the file [{path}]")
+
+    def get_view_from_cache(self, view_name: ViewName) -> Optional[LookMlView]:
+        """
+        Check if view is cached, and return it.
+        Otherwise, return None
+        """
+        if view_name in self._views_cache:
+            return self._views_cache[view_name]
+
+        return None
+
+    def find_view(self, view_name: ViewName, path: Includes) -> Optional[LookMlView]:
```

Comment:
I don't think this method is being called, and the `path` argument is unused

Reply:
@pmbrull I've remove the `path` argument, btw, this method is still called in `_process_view()` function
https://github.com/open-metadata/OpenMetadata/blob/60633dbfc0dc9ad319f2cddff4a4496ddcb5d2f1/ingestion/src/metadata/ingestion/source/dashboard/looker/metadata.py#L526